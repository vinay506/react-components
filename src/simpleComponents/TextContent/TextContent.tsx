import React from 'react'
import styled from 'styled-components';
function TextContent(props: any) {
    /**
     * Text: The text which should be displayed on web page
     * color: Defaults to darkgrey, To override the existing color we can pass Color code or color name as props.
     */
    const { text, color } = props;
    const customStyles = {
        color: color
    };
    return (
        <React.Fragment>
            <Label id="textContent" style={customStyles}>
                {text}
            </Label>
        </React.Fragment >
    )
}

const Label = styled.div`
   color: darkgrey;
    font-size: 13px;
    font-family: 'AmazonEmber-Regular';
     letter-spacing: 0;
  line-height: 20px;
    `;
export default TextContent
