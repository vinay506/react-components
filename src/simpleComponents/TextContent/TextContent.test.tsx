import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import TextContent from './TextContent';
import 'jest-styled-components';
describe('Text Content', () => {
    let wrapper: any;
    let propInfo: any;
    let defaultStyles: any;
    beforeEach(() => {
        propInfo = {
            text: 'Sample Text Primary',
            color: 'red'
        };
        defaultStyles = {
            color: 'darkgrey',
            fontSize: '13px',
            fontFamily: 'AmazonEmber-Regular',
            letterSpacing: '0',
            lineHeight: '20px'
        }
        wrapper = shallow(<TextContent {...propInfo} />);
    });
    test('Content Text Snapshot', () => {
        const instance = renderer.create(<TextContent {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('Content Text Element check', () => {
        const elementLength = wrapper.find('#textContent').length;
        expect(elementLength).toEqual(1);
    })
    test('Content Text Validation', () => {
        const primaryText = wrapper.find('#textContent').text();
        expect(primaryText).toEqual(propInfo.text);
    })
    test('Content Text Color Validation', () => {
        const primaryTextStyles = wrapper.find('#textContent');
        expect(primaryTextStyles.props().style.color).toEqual(propInfo.color);
    })

    test('Text Content Default Color Validation', () => {
        const instance = renderer.create(<TextContent {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.color);
    })

    test('Text Content font-size Validation', () => {
        const instance = renderer.create(<TextContent {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontSize);
    })

    test('Text Content font-family Validation', () => {
        const instance = renderer.create(<TextContent {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.fontFamily);
    })

    test('Text Content letterSpacing Validation', () => {
        const instance = renderer.create(<TextContent {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.letterSpacing);
    })

    test('Text Content lineHeight Validation', () => {
        const instance = renderer.create(<TextContent {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toHaveStyleRule(defaultStyles.lineHeight);
    })

})

