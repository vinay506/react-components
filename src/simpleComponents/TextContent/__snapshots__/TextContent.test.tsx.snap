// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Content Content Text Snapshot 1`] = `
.c0 {
  color: darkgrey;
  font-size: 13px;
  font-family: 'AmazonEmber-Regular';
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 20px;
}

<div
  className="c0"
  id="textContent"
  style={
    Object {
      "color": "red",
    }
  }
>
  Sample Text Primary
</div>
`;
